packages/vscode-extension/src/debug/teamsfxTaskHandler.ts

```diff
@@ -168,6 +226,12 @@ async function onDidEndTaskProcessHandler(event: vscode.TaskProcessEndEvent): Pr
   const task = event.execution.task;
   const activeTerminal = vscode.window.activeTerminal;
 
+  const taskId = task?.definition?.teamsfxTaskId;
+  if (taskId !== undefined) {
+    trackedTasks.delete(taskId as string);
+    taskEndEventEmitter.fire({ id: taskId as string, exitCode: event.exitCode });
```

Comment:
Curious what is the behavior when a task end listener is fired twice, i.e., one on onDidEndTaskProcessHandler and the other on onDidEndTaskHandler?

Reply:
Per vscode source code, `onDidEndTaskProcessHandler` will be called first. For normal case `trackedTasks` is updated so `onDidEndTaskHandler` will not fire event.

For error case that process has not been started, only `onDidEndTaskHandler` will fire event.

For corner case that end event is fired twice, updated `runTask` to dispose listener after receiving the first event. So the second event has no effect on `runTask`.